assets/global.js

```diff
@@ -533,10 +541,17 @@ class VariantSelects extends HTMLElement {
     const newMedia = document.querySelector(
       `[data-media-id="${this.dataset.section}-${this.currentVariant.featured_media.id}"]`
     );
+    const modalContent = document.querySelector('.product-media-modal__content');
+    const newMediaModal = modalContent.querySelector( `[data-media-id="${this.currentVariant.featured_media.id}"]`);
+
     if (!newMedia) return;
     const parent = newMedia.parentElement;
+    if (parent.firstChild == newMedia) return;
+    modalContent.prepend(newMediaModal);
     parent.prepend(newMedia);
-    window.setTimeout(() => { parent.scroll(0, 0) });
+    parent.dispatchEvent(this.mediaChangeEvent);
+    window.setTimeout(() => { parent.scrollLeft = 0; parent.scrollIntoView({behavior: "smooth", inline: "start"}); });
+    window.setTimeout(() => { this.header.productMediaIsChanging = false;}, 800);
```

Comment:
The 800ms does seem reliable enough, but obviously it's a little fragile because it’s at the mercy of each browser's scrollintoview speed, which we can’t control.

If it bothers anyone else, we could consider removing this bit and using a "scrolling is finished" timeout check in the existing onScroll method in header.liquid and reset the `productMediaIsChanging` flag there. But otherwise I'm ok with this for now.

Reply:
Hey @kmeleta, i've updated it here, let me know if you see anything I should/could do differently: https://github.com/Shopify/dawn/pull/158/commits/224dced7e7aae5fdee1d12a156ce2e2517e21ab0